src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -827,7 +839,10 @@ static int enumerate_partitions(dev_t devnum) {
                         r = k;
         }
 
-        if (m->partitions[PARTITION_ESP].found) {
+        if (m->partitions[PARTITION_ESP].found &&
+            esp_is_valid(m->partitions + PARTITION_ESP,
+                         m->partitions[PARTITION_XBOOTLDR].uuid)) {
```

Comment:
this really should check `m->partitions[PARTITION_XBOOTLDR].found` before using `m->partitions[PARTITION_XBOOTLDR].uuid`